plugins/inventory/inventory.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2022, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+name: inventory
+author:
+  - Polona Mihalič (@PolonaM)
+short_description: Inventory source from Terraform state file.
+description:
+  - Builds an inventory from specified state file.
+  - Does not support caching.
+version_added: 1.0.0
+seealso: []
+options:
+  plugin:
+    description:
+      - The name of the Inventory Plugin.
+      - This should always be C(cloud.terraform.inventory).
+    required: true
+    type: str
+    choices: [ cloud.terraform.inventory ]
+    version_added: 1.0.0
+  project_path:
+    description:
+      - The path to the root of the Terraform directory with the .tfstate file.
+      - If I(state_file) and I(project_path) are not specified, C(terraform.tfstate) file in the current
+        working directory will be used.
+    type: path
+    version_added: 1.0.0
+  state_file:
+    description:
+      - Absolute path to an existing Terraform state file whose outputs will be listed.
+      - If I(state_file) and I(project_path) are not specified, C(terraform.tfstate) file in the current
+        working directory will be used.
+    type: path
+    version_added: 1.0.0
+  binary_path:
+    description:
+      - The path of a terraform binary to use.
+    type: path
+    version_added: 1.0.0
+"""
+
+EXAMPLES = r"""
+# Example that creates an inventory from terraform.tfstate file in cwd
+
+plugin: cloud.terraform.inventory
+
+
+# Example that creates an inventory from mycustomstate.tfstate file in selected project directory
+
+plugin: cloud.terraform.inventory
+project_path: some/project/path
+state_file: mycustomstate.tfstate
+"""
+
+
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+import yaml
+import json
+import subprocess
+from ansible.errors import AnsibleError
+from ansible.module_utils.common import process
+import os
+
+
+# TODO: remove when ansible provider is available
+TESTING_STATE_FILE = os.environ.get('TESTING_STATE_FILE', False)
```

Comment:
Make this a string envvar containing the path to the file, not a boolean. Also, this is a state _show_ file :)

Reply:
I used it as a flag if state file for test (show file) needs to be used. Path to the show file is defined further in the code.